tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/Samples/SampleGeneratorTool.cs

```diff
@@ -0,0 +1,227 @@
+
+        public override async Task<CommandResponse> HandleCommand(ParseResult parseResult, CancellationToken ct)
+        {
+            // Retrieve raw prompt argument
+            string rawPrompt = parseResult.GetValue(promptOption) ?? "";
+            string prompt = rawPrompt;
+
+            // If the raw prompt looks like a path to a markdown file (no newlines, ends with .md/.markdown, file exists), load its content.
+            if (!string.IsNullOrWhiteSpace(rawPrompt)
+                && rawPrompt.IndexOf('\n') == -1
+                && rawPrompt.IndexOf('\r') == -1)
+            {
+                try
+                {
+                    var trimmed = rawPrompt.Trim();
+                    if (trimmed.EndsWith(".md", StringComparison.OrdinalIgnoreCase) || trimmed.EndsWith(".markdown", StringComparison.OrdinalIgnoreCase))
+                    {
+                        var fullPath = Path.GetFullPath(trimmed);
+                        if (File.Exists(fullPath))
+                        {
+                            logger.LogInformation("Loading prompt content from file: {promptFile}", fullPath);
+                            prompt = await File.ReadAllTextAsync(fullPath, ct);
+                        }
+                    }
+                }
+                catch (Exception ex)
+                {
+                    logger.LogWarning(ex, "Failed to read prompt file '{file}'. Falling back to raw prompt text.", rawPrompt);
+                    prompt = rawPrompt; // fallback
+                }
+            }
+            string packagePath = parseResult.GetValue(SharedOptions.PackagePath) ?? ".";
+            bool overwrite = parseResult.GetValue(overwriteOption);
+            var model = parseResult.GetValue(modelOption);
+
+            try
+            {
+                await GenerateSampleAsync(prompt, packagePath, overwrite, model, ct);
+                return new DefaultCommandResponse { Message = "Sample generation completed successfully" };
+            }
+            catch (ArgumentException ex)
+            {
+                logger.LogError(ex, "SampleGenerator failed with validation errors");
+                return new DefaultCommandResponse { ResponseError = $"SampleGenerator failed with validation errors: {ex.Message}" };
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex, "SampleGenerator threw an exception");
+                return new DefaultCommandResponse { ResponseError = $"SampleGenerator threw an exception: {ex.Message}" };
+            }
+        }
+
+        private async Task GenerateSampleAsync(string prompt, string packagePath, bool overwrite, string model, CancellationToken ct)
+        {
+            IPackageInfoHelper? helper = await packageInfoResolver.Resolve(packagePath, ct) ?? throw new ArgumentException("Unable to determine language for package (resolver returned null). Ensure repository structure and Language-Settings.ps1 are correct.");
```

Comment:
I don't understand the need for PackageInfoHelper. Feel like most of it could be resolved by options or inferred from user context of package path + options?